scripts/conflicts/diff.js

```diff
@@ -20,37 +26,90 @@ if (!server || argv.h || argv.help) {
-              // console.log(results);
-              // Make sure mine and theirs are the right way around
-              const mine = results.find(r => r.ok._rev !== conflictRev).ok;
-              const theirs = results.find(r => r.ok._rev === conflictRev).ok;
-
-              console.log('=====================');
-              console.log(r.id);
-              console.log('=====================');
-              console.log(jsondiff.diffString(mine, theirs));
-              console.log('\n\n');
-            });
-        }
+        const docId = r.id;
+        const arrayRevsIdThatShouldNotBeThere = r.key;
+        return DB.get(docId, { open_revs: 'all' })
+          .then(results => {
+
+            // Get the main_doc. Maindoc is the doc that isn't in the conflict list
+            const main_doc = results.filter(r => !arrayRevsIdThatShouldNotBeThere.includes(r.ok._rev))[0].ok; 
+
+            //Now we are going to put each doc, its conflicts and thier diff 
+            //in one file for manual inspection if required.
+            //The doc id might have characters that cannot be used to create folders.
+            let conflictDirectoryName = r.id;
+            if(r.id.includes('target')){
+              conflictDirectoryName = 'target_' + r.id.split(':')[0];
+            } else if (r.id.includes('task')){ //If doc is task add ask and part of doc id
+              conflictDirectoryName = 'task_' + r.id.split(':')[1];
+            } else {  // if it is just a contact doc use the doct Id to create folder
+              conflictDirectoryName = r.id;
+            }
+            
+            conflictDirectoryName.replace(/[^0-9a-z]/gi, '');
+            const conflictDirectoryPath = `${mainConflictFilesDirectory}/${conflictDirectoryName}`; 
+            if (!fs.existsSync(conflictDirectoryPath)){
+              try {
+                fs.mkdirSync(conflictDirectoryPath);
+              } catch(err){
+                console.log(err);
+              }
+            }
+
+            writeToFile(conflictDirectoryPath + '/' + 'mainDocument.json', main_doc);
+            r.key.forEach( conflict_doc_id => {
+              const conflict_doc = results.find(r => r.ok._rev === conflict_doc_id).ok;
+              //Generate Json Diff
+              const jsonDiff = jsondiff.diff(main_doc, conflict_doc);
+              //Generate String Diff
+              const stringJsonDiff = jsondiff.diffString(main_doc, conflict_doc);
+              console.log(stringJsonDiff);
+
+              //Write JSON diff between the conflict files
+              writeToFile(conflictDirectoryPath +'/diff_' +conflict_doc_id +'.json', jsonDiff);
+
+              //Write conflict doc to the drive
+              writeToFile(conflictDirectoryPath +'/'+ conflict_doc_id + '.json', conflict_doc);
+
+              //Write string diff between the conflict files
+              writeToFile(conflictDirectoryPath + '/string_diff_' + conflict_doc_id + '.txt', stringJsonDiff);
```

Comment:
The string diff looks super weird when written to the file: 
```
" {\n\u001b[31m-  _rev: \"8-de3fe258b9e30b7096b9125ce1f2276d\"\u001b[39m\n\u001b[32m+  _rev: \"4-d50f73817a473255c40e6b06eebbe78b\"\u001b[39m\n\u001b[31m-  facility_id: \"c5d73dde-c62b-4297-a78f-929e59438686\"\u001b[39m\n\u001b[32m+  facility_id: \"dfe8e4d8-5338-4321-8751-e2ee64ec5bbf\"\u001b[39m\n\u001b[31m-  contact_id: \"afa1cfabc855af4abd7a9721dd001624\"\u001b[39m\n\u001b[32m+  contact_id: \"50472724-b83f-4ce8-b7f7-58f267a5442f\"\u001b[39m\n }\n"
```

Is it an option to save the diff as json? Or is this text format more useful somehow?

Reply:
I have put out both the string and JSON diff. The String diff was there from the beginning so I didn't change it.

Reply:
Removed string diff. Json diff had more and clearer content. Thank you for suggestion.